src/rcc.rs

```diff
@@ -967,9 +1051,35 @@ impl CFGR {
             while rcc.csr.read().lsirdy().is_not_ready() {}
         }
 
-        if self.use_pll48clk {
-            // set source clock for 48 MHz to main PLL
-            rcc.dckcfgr2.modify(|_, w| w.ck48msel().bit(false));
+        if self.use_pllsai {
+            let pllsain_freq = match self.hse.as_ref() {
+                Some(hse) => hse.freq.integer() as u64 / self.pllm as u64 * self.pllsain as u64,
+                None => 16_000_000 / self.pllm as u64 * self.pllsain as u64,
+            };
+            let pllsaip_freq = pllsain_freq
+                / match self.pllsaip {
+                    PLLSAIP::Div2 => 2,
+                    PLLSAIP::Div4 => 4,
+                    PLLSAIP::Div6 => 6,
+                    PLLSAIP::Div8 => 8,
+                };
+            // let pllsaiq_freq = pllsain_freq / self.pllsaiq as u64;
+            assert!((192_000_000..=432_000_000).contains(&pllsain_freq));
```

Comment:
I'm not sure here, the reference manual (RM0410 Rev 4, Page 212), says the following `Caution: The software has to set these bits correctly to ensure that the VCO output frequency
is between 100 and 432 MHz.`, but STM32CubeMX states 192 MHz as the minimum.

Reply:
Could you please add that exact comment as a comment to the code?